src/main/java/teammates/common/datatransfer/questions/FeedbackMcqResponseDetails.java

```diff
@@ -77,7 +77,21 @@ public String getAnswerCsv(FeedbackQuestionDetails questionDetails) {
 
     @Override
     public List<String> validateResponseDetails(FeedbackQuestionAttributes correspondingQuestion) {
-        return new ArrayList<>();
+        List<String> errors = new ArrayList<>();
+        List<String> mcqChoices = ((FeedbackMcqQuestionDetails) correspondingQuestion
+                .getQuestionDetails()).getMcqChoices();
+
+        // if other option is not selected and selected answer is not part of Mcq option list trigger this error.
+        if (!isOther && !mcqChoices.contains(getAnswerString())) {
+            errors.add(getAnswerString() + " " + Const.FeedbackQuestion.MCQ_ERROR_INVALID_OPTION);
+        }
+
+        // if other option is selected but not text is provided trigger this error
+        if (isOther && getAnswerString().trim().equals("")) {
```

Comment:
Should the user be allowed to submit a feedback if `isOther` is selected and not text is provided for the other option?

Reply:
What should be the expected behavior for this?

Reply:
If you want to prohibit empty string as answer, this should be no different.

Reply:
Got it. 👍